### DISCUSSION TITLE
TinyMCE does not behave intuitively when in a FlexBox

### DISCUSSION THREAD

**Discussion by jcobban** [tinymce/tinymce#8787] [Ideas]
I have noticed that when TinyMCE is used to replace a `<textarea>` that is contained inside a tag with `display:flex` it does not display intuitively in my opinion.  For example if I take a page which without TinyMCE displays as:

![image](https://github.com/tinymce/tinymce/assets/17891982/09225f78-eac7-416d-a006-af58d529c7d6)

It appears as:

![image](https://github.com/tinymce/tinymce/assets/17891982/869f1415-b6c8-4324-baeb-53d63ba802dd)

Observe that the editor only occupies 1/4th of the width of the containing tag.  In this case the editor is enclosed by a div with display: flex.  If I remove the display: flex and put in width: 80% the edit displays as:

![image](https://github.com/tinymce/tinymce/assets/17891982/e6ab2781-d021-4c2a-9abd-b64c7c4f9d3f)

This is not as easy to resolve in situations where display: flex is desirable for layout.

```
        <div class="row" id="notesRow">
          <label class="column1" for="note">
                Notes:
          </label>
          <textarea name="note" id="note" 
                    cols="64" rows="10" class="$notesClass"
              >$notes</textarea>
        </div>

...
    div.row {   white-space:        nowrap;
                width:              100%;
                display:            flex;
            }

```
 Without TinyMCE this displays as: 

![image](https://github.com/tinymce/tinymce/assets/17891982/8241b948-19a1-406c-afc7-0f93040cfc16)

but with TinyMCE it displays as:

![image](https://github.com/tinymce/tinymce/assets/17891982/987a18a7-fe09-4073-a117-2d4d72226cfe)

To get around this I have to enclose the `<textarea>` in a tag with  `style="flex-grow: 1;"` which convinces the browser to display the editor in the full remaining width of the flex-box.

```
        <div class="row" id="notesRow">
          <label class="column1" for="note">
                Notes:
          </label>
          <span style="flex-grow: 1;">
          <textarea name="note" id="note" 
                    cols="64" rows="10" class="$notesClass"
              >$notes</textarea>
          </span>
```

![image](https://github.com/tinymce/tinymce/assets/17891982/97c59306-b1a6-4e60-a067-ef90f3d8f931)

Again the editor only occupies 1/4th of the width of the containing tag.

In all cases the TinyMCE editor ignores the width hint from the `<textarea>`.

FYI this is in my family history website https://www.jamescobban.net.  However I only use TinyMCE in portions of the website which require signing on.  Accounts and all services on the site are free.  I used TinyMCE to create several blog posts at https://www.jamescobban.net/Blogs.php

**Comment by lostkeys** [comment]
@jcobban your solution is probably the right one. TinyMCE does not use any styles applied to the textarea. Another approach is to inspect the code to get the TinyMCE root element and add your own styling to that if needed.
